data/csv.go

```diff
@@ -0,0 +1,32 @@
+package data
+
+import (
+	"encoding/csv"
+	"log"
+	"os"
+)
+
+type csvDataSource struct {
+	// Collection is a file path to a CSV file
+	Collection string
+}
+
+func NewCSVDataSource(csvPath string) csvDataSource {
+	return csvDataSource{Collection: csvPath}
+}
+
+func (ds csvDataSource) ReadCollection() [][]string {
+	file, err := os.Open(ds.Collection)
+	if err != nil {
+		log.Fatalln(err)
```

Comment:
I agree with you, at first I thought the program should exit fatally ASAP, because if it can't read the CSV, the API wouldn't even work.

But as you say, I think a better approach would be to propagate the error up, to let the caller decide how to handle it and not just immediately exit.